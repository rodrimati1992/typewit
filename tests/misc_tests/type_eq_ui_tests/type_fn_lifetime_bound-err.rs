typewit::type_fn!{
    struct WithLts<'a, 'b: 'a, 'c: 'a + 'static, 'd: 'static + 'a>;

    impl () => ()
}

fn _with_lifetime<'a, 'b, 'c, 'd>() {
    let _: WithLts<'a, 'b, 'c, 'd>;
}

fn main() {}