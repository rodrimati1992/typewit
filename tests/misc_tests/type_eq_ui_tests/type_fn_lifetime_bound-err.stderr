error: lifetime may not live long enough
 --> tests/misc_tests/type_eq_ui_tests/type_fn_lifetime_bound-err.rs:8:12
  |
7 | fn _with_lifetime<'a, 'b, 'c, 'd>() {
  |                   --  -- lifetime `'b` defined here
  |                   |
  |                   lifetime `'a` defined here
8 |     let _: WithLts<'a, 'b, 'c, 'd>;
  |            ^^^^^^^^^^^^^^^^^^^^^^^ requires that `'b` must outlive `'a`
  |
  = help: consider adding the following bound: `'b: 'a`

error: lifetime may not live long enough
 --> tests/misc_tests/type_eq_ui_tests/type_fn_lifetime_bound-err.rs:8:12
  |
7 | fn _with_lifetime<'a, 'b, 'c, 'd>() {
  |                           -- lifetime `'c` defined here
8 |     let _: WithLts<'a, 'b, 'c, 'd>;
  |            ^^^^^^^^^^^^^^^^^^^^^^^ requires that `'c` must outlive `'static`

error: lifetime may not live long enough
 --> tests/misc_tests/type_eq_ui_tests/type_fn_lifetime_bound-err.rs:8:12
  |
7 | fn _with_lifetime<'a, 'b, 'c, 'd>() {
  |                               -- lifetime `'d` defined here
8 |     let _: WithLts<'a, 'b, 'c, 'd>;
  |            ^^^^^^^^^^^^^^^^^^^^^^^ requires that `'d` must outlive `'static`

help: the following changes may resolve your lifetime errors
  |
  = help: add bound `'b: 'a`
  = help: replace `'c` with `'static`
  = help: replace `'d` with `'static`
